### PR TITLE
(feat) added 0xframe and coinbase wallet

### PR DESCRIPTION
**Pull request by karmabadger**
This does not add walletconnect which is the main issue with webpack.

### CODE REVIEW

**Comment by karmabadger** [comment]
This seems to work locally though

```
yarn build00  11:47:26  
yarn run v1.22.17
$ react-scripts build
Creating an optimized production build...

warn - Tailwind is not purging unused styles because no template paths have been provided.
warn - If you have manually configured PurgeCSS outside of Tailwind or are deliberately not removing unused styles, set `purge: false` in your Tailwind config file to silence this warning.
warn - https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
Compiled with warnings.

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\@metamask\safe-event-emitter\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\@metamask\safe-event-emitter\index.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\classes.ts' file: Error: ENOENT: no such file or directory, open 
'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\classes.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\error-constants.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\error-constants.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\errors.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\errors.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\index.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\utils.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\eth-rpc-errors\src\utils.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\JsonRpcEngine.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\JsonRpcEngine.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\createAsyncMiddleware.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\createAsyncMiddleware.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\createScaffoldMiddleware.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\createScaffoldMiddleware.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\getUniqueId.ts' file: Error: ENOENT: no such file or directory, 
open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\getUniqueId.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\idRemapMiddleware.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\idRemapMiddleware.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\index.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\mergeMiddleware.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\eth-json-rpc-filters\node_modules\json-rpc-engine\src\mergeMiddleware.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\classes.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\classes.ts' 

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\error-constants.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\error-constants.ts'

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\errors.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\errors.ts'   

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\index.ts'     

Failed to parse source map from 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\utils.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\wenxu\Projects\engineer-dao\platform\node_modules\walletlink\node_modules\eth-rpc-errors\src\utils.ts'     

src\components\wallet\useWalletProvider.ts
  Line 15:7:  'infuraId' is assigned a value but never used  @typescript-eslint/no-unused-vars

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  643.81 kB (+137.14 kB)  build\static\js\main.5bc33023.js
  544.55 kB               build\static\css\main.ef23a2c1.css
  1.67 kB (+1 B)          build\static\js\512.6be25c57.chunk.js

The bundle size is significantly larger than recommended.
Consider reducing it with code splitting: https://goo.gl/9VhYWB
You can also analyze the project dependencies: https://goo.gl/LeUzfb

The project was built assuming it is hosted at /.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.
You may serve it with a static server:

  npm install -g serve
  serve -s build

Find out more about deployment here:

  https://cra.link/deployment

Done in 199.54s.
```

**Comment by karmabadger** [comment]
I did add some env vars so might be that? they aren't added to the vercel deployment right?

**Comment by karmabadger** [comment]
I will add wallectconnect once this part works and is merged

File: frontend/src/components/wallet/useWalletProvider.ts
Comment (reisr3) [inline frontend/src/components/wallet/useWalletProvider.ts:13]: Use imports instead of require

**Comment by reisr3** [comment]
@catmaxi You have a lint error:

` Line 15:7:  'infuraId' is assigned a value but never used  @typescript-eslint/no-unused-vars`

File: frontend/src/components/wallet/useWalletProvider.ts
Comment (reisr3) [inline frontend/src/components/wallet/useWalletProvider.ts:36]: Don't comment out, just remove.

File: frontend/src/components/wallet/useWalletProvider.ts
Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts:13]: there are no types provided for this package yet and I'm not very familiar with that part as I'm a bit new with TS type declarations, a bit of help would be greatly appreciated thanks :)

File: frontend/src/components/wallet/useWalletProvider.ts
Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts]: removed

File: frontend/src/components/wallet/useWalletProvider.ts
Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts]: ok so I added a module type declaration file for 'eth-provider' but I'm really not sure if this is the right way to do it. Even tho it does seem to work even if it doesn't actually provide any types.

Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts]: at frontend/src/types/eth-provider.d.ts:
```ts
declare module 'eth-provider';
```

**Comment by karmabadger** [comment]
made the appropriate changes

**Comment by karmabadger** [comment]
again I'll wait for this one to be merged before adding in the polyfills and walletconnect since they kind of depend on this part first I think

**Comment by Yuxanju** [comment]
You will have to synchronize your wallet and make sure you debug it with bech13.

File: frontend/.env.example
Comment (deweller) [inline frontend/.env.example:5]: Is this a live API key?

**Review by deweller — CHANGES_REQUESTED** [review]
Unfortunately, it doesn't look like Coinbase wallet is working with Webpack 5 either:

image

We'll need to sort all this out with WalletConnect as well.

**Comment by Yuxanju** [comment]
You will need to synchronize your wallet and then you debug it

**Comment by karmabadger** [comment]
> You will have to synchronize your wallet and make sure you debug it with bech13.

wdym?

**Comment by karmabadger** [comment]
> Unfortunately, it doesn't look like Coinbase wallet is working with Webpack 5 either:
> 
> image
> 
> We'll need to sort all this out with WalletConnect as well.

hmmm this does seem to be a polyfills problem, what were the steps to reproduce that?

**Comment by karmabadger** [comment]
oh well at this point I think I will add the polyfills for this branch anyways. The most important wallet out of those to be added is still walletconnect as multisig wallets like gnosis safe can use it so this would certainly be worth it. 

adding polyfills should be the right option as I don't think that downgrading to webpack 4 is a good idea. Not exactly sure why webpack decided all of a sudden that not automatically doing these polyfills is a good idea since it would probably cause a significant amount of repos to break but yeah.

**Comment by karmabadger** [comment]
ok so I just added polyfills, yarn build seems to be working on my end.
there are a few new packages to install and a few ```Failed to parse source map from...``` warnings from builds which aren't important but might cause an error with the deployments depending on the configs I guess?

all react-scripts scripts need to be changed to ```react-app-rewired``` though, for instance, ```react-app-rewired build```.

react-scripts will definitely not work as it does not come with the right polyfills.
At this point, the modals work correctly and the connection of the wallet works correctly as per commit #042fdd039bf4a409357e66ebda71535c956b2a0a
However, there seems to still be some issues with rpc urls connections.

**Comment by indeavr** [comment]
Hey @karmabadger could you please squash your commits in a single one ? It'll make reviewing much easier. 
Also, @deweller what is keeping this from being merged ?

**Comment by reisr3** [comment]
@karmabadger Let's revisit this after launch! Closing for now.

**Pull request closed without merging**
